File: c_and_cpp/string_examples.cpp
#include <iostream>

int main(int argc, const char * argv[]) {

  // Make a string
  std::string s("Hello");

  // Find length of a string
  std::cout << s.length() << std::endl;

  // Compare two strings
  std::string s2("Hello 2");
  std::cout << (s == s2) << std::endl;

  // Search for a substring
  s2 = std::string("el");
  std::cout << s.find(s2) << std::endl;

  // Make a substring
  size_t start = 1;
  size_t end = 3;
  s2 = s.substr(start, end);
  std::cout << s2 << std::endl;

  // Split a string


  return 0;
}

File: Java/src/net/jimblacker/questions/StringExamples.java
package net.jimblacker.questions;

public class StringExamples {
  public static void main(String[] args) {

    // Make a string
    String s = "Hello";

    // Find length of a string
    System.out.println(s.length());

    // Compare two strings
    String s2 = "Hello 2";
    System.out.println(s.equals(s2));

    // Search for a substring
    s2 = "el";
    System.out.println(s.indexOf(s2));

    // Make a substring
    int b = 1;
    int e = 3;
    s2 = s.substring(b, e);
    System.out.print(s2);
  }

}
